pkg/api/types.go

```diff
@@ -827,6 +833,16 @@ type ObjectFieldSelector struct {
 	FieldPath string `json:"fieldPath"`
 }
 
+// ResourceFieldSelector represents container resources (cpu, memory) and their output format
+type ResourceFieldSelector struct {
+	// Container name
```

Comment:
Add a note saying this is required if used in the context of a volume, optional for container.